Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogHeaderWidget.cpp

```diff
@@ -187,82 +194,126 @@ namespace O3DE::ProjectManager
+        downloadProgessBar->setObjectName("DownloadProgressBar");
+        downloadingGemLayout->addWidget(downloadProgessBar);
+        downloadProgessBar->setValue(0);
+
+        m_downloadingListWidget->layout()->addWidget(newGemDownloadWidget);
+
+        const AZStd::vector<QString>& downloadQueue = m_downloadController->GetDownloadQueue();
+        QLabel* numDownloads = m_downloadingListWidget->findChild<QLabel*>("NumDownloadsInProgressLabel");
+        numDownloads->setText(QString("%1 %2")
+                                  .arg(downloadQueue.size())
+                                  .arg(downloadQueue.size() == 1 ? tr("download in progress...") : tr("downloads in progress...")));
+
+        m_downloadingListWidget->show();
+    }
 
-                for (int downloadingGemNumber = 0; downloadingGemNumber < downloadQueue.size(); ++downloadingGemNumber)
-                {
-                    QHBoxLayout* nameProgressLayout = new QHBoxLayout();
-                    TagWidget* newTag = new TagWidget(downloadQueue[downloadingGemNumber]);
-                    nameProgressLayout->addWidget(newTag);
-                    QLabel* progress = new QLabel(downloadingGemNumber == 0? QString("%1%").arg(downloadProgress) : tr("Queued"));
-                    nameProgressLayout->addWidget(progress);
-                    QSpacerItem* spacer = new QSpacerItem(0, 0, QSizePolicy::Expanding, QSizePolicy::Minimum);
-                    nameProgressLayout->addSpacerItem(spacer);
-                    QLabel* cancelText = new QLabel(QString("<a href=\"%1\">Cancel</a>").arg(downloadQueue[downloadingGemNumber]));
-                    cancelText->setTextInteractionFlags(Qt::LinksAccessibleByMouse);
-                    connect(cancelText, &QLabel::linkActivated, this, &CartOverlayWidget::OnCancelDownloadActivated);
-                    nameProgressLayout->addWidget(cancelText);
-                    downloadingItemLayout->addLayout(nameProgressLayout);
-                    QProgressBar* downloadProgessBar = new QProgressBar();
-                    downloadingItemLayout->addWidget(downloadProgessBar);
-                    downloadProgessBar->setValue(downloadingGemNumber == 0 ? downloadProgress : 0);
-                }
+    void CartOverlayWidget::GemDownloadRemoved(const QString& gemName)
+    {
+        QWidget* gemToRemove = m_downloadingListWidget->findChild<QWidget*>(gemName);
+        QLayout* gemToRemoveLayout = gemToRemove ? gemToRemove->layout() : nullptr;
 
-                widget->setUpdatesEnabled(true);
-                widget->show();
+        if (gemToRemoveLayout)
+        {
+            QLayoutItem* rowLayoutItem = nullptr;
+            while ((rowLayoutItem = gemToRemoveLayout->layout()->takeAt(0)) != nullptr)
+            {
+                rowLayoutItem->widget()->deleteLater();
             }
-        };
+            gemToRemoveLayout->layout()->deleteLater();
+            gemToRemove->deleteLater();
```

Comment:
Do you need to delete each individual widget? I thought just deleting the parent widget (gemToRemove) itself would delete all of its children as well? Do the widgets in its layout have gemToRemove set as their parent?

Reply:
In this case they did because none of the child widgets had a parent set, but there's no reason why that was the case. I've set the parent of all child widgets to the container so this could be replaced with one delete.